static/app/views/dashboards/editAccessSelector.tsx

```diff
@@ -0,0 +1,196 @@
+import React, {useState} from 'react';
+import styled from '@emotion/styled';
+import isEqual from 'lodash/isEqual';
+
+import AvatarList from 'sentry/components/avatar/avatarList';
+import TeamAvatar from 'sentry/components/avatar/teamAvatar';
+import Badge from 'sentry/components/badge/badge';
+import {CompactSelect} from 'sentry/components/compactSelect';
+import {CheckWrap} from 'sentry/components/compactSelect/styles';
+import UserBadge from 'sentry/components/idBadge/userBadge';
+import {InnerWrap, LeadingItems} from 'sentry/components/menuListItem';
+import {Tooltip} from 'sentry/components/tooltip';
+import {t} from 'sentry/locale';
+import type {Team} from 'sentry/types/organization';
+import type {User} from 'sentry/types/user';
+import {useTeamsById} from 'sentry/utils/useTeamsById';
+import {useUser} from 'sentry/utils/useUser';
+import type {DashboardDetails, DashboardPermissions} from 'sentry/views/dashboards/types';
+
+interface EditAccessSelectorProps {
+  dashboard: DashboardDetails;
+  onChangeEditAccess?: (newDashboardPermissions?: DashboardPermissions) => void;
+}
+
+/**
+ * Dropdown multiselect button to enable selective Dashboard editing access to
+ * specific users and teams
+ */
+function EditAccessSelector({dashboard, onChangeEditAccess}: EditAccessSelectorProps) {
```

Comment:
Here, `dashboard` isn't a state variable. This means that when the perms are updated, it doesn't instantly update the overall dashboard's state. However, I 'locally' manage the perms state within the component (line 35). 

I figured this would be okay since only the creator will be editing perms and it doesn't need to instantly reflect on the dashboard. Is this okay?